lib/openssl/ssl.rb

```diff
@@ -459,6 +459,32 @@ def session
         nil
       end
 
+      # Close the stream for reading.
+      # This method is ignored by OpenSSL as there is no reasonable way to
+      # implement it, but exists for compatibility with IO.
+      def close_read
+        # Unsupported and ignored.
+        # Just don't read any more.
+      end
+
+      # Closes the stream for writing. The behavior of this method depends on
+      # the version of OpenSSL and the TLS protocol in use.
+      # 
+      # In TLS 1.3 and later:
+      # - Sends a 'close_notify' alert to the peer.
+      # - Does not wait for the peer's 'close_notify' alert in response.
+      # 
+      # In TLS 1.2 and earlier:
+      # - Sends a 'close_notify' alert to the peer.
+      # - Waits for the peer's 'close_notify' alert in response.
```

Comment:
This `#close_write` method does the same thing in both protocol versions. It sends a `close_notify` and returns without waiting for the response.

How the peer will react to the `close_notify` is different depending on the protocol version.